src/main/java/macrobase/analysis/transform/SlidingWindowTransform.java

```diff
@@ -0,0 +1,67 @@
+package macrobase.analysis.transform;
+
+import macrobase.analysis.pipeline.stream.MBStream;
+import macrobase.analysis.transform.aggregate.BatchedWindowAggregate;
+import macrobase.conf.MacroBaseConf;
+import macrobase.conf.ConfigurationException;
+
+import macrobase.conf.MacroBaseDefaults;
+import macrobase.datamodel.Datum;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class SlidingWindowTransform extends FeatureTransform {
+    public int minPaneSize;
+    public BatchedWindowAggregate windowAggregate;
```

Comment:
Put another way, if every aggregate implements `IncrementalWindowAggregate`, then all this function needs to do is keep track of panes and when they enter and expire...